### DISCUSSION TITLE
How to override the default rpcUrl when use WagmiAdapter from @reown/appkit-adapter-wagmi?

### DISCUSSION THREAD

**Discussion by juli2a** [WalletConnect/walletconnect-monorepo#5512] [Developer Support] [answered]
My project is built with: React, Wagmi, AppKit.
I am trying to add transports to the WagmiAdapter config, as described in the Wagmi documentation for the createConfig method, but it doesn't give results. I see in console the default rpcUrl from WalletConnect:

![pic1](https://github.com/user-attachments/assets/985d135c-657a-4e30-9fa1-df0217dae853)

Are there any ways to override default rpcUrl?

Here's my code:

```
import { http, WagmiProvider} from 'wagmi'
import { createAppKit } from '@reown/appkit/react'
import { sepolia, polygonAmoy} from '@reown/appkit/networks'
import { WagmiAdapter } from '@reown/appkit-adapter-wagmi'

const queryClient = new QueryClient()

const generalConfig = {
  projectId: myProjectId,
  metadata: myMetadata,
  networks: [sepolia, polygonAmoy],
}

const wagmiAdapter = new WagmiAdapter({
  ...generalConfig,
  transports: {
    [sepolia.id]: http(`https://sepolia.infura.io/v3/${INFURA_KEY}`),
    [polygonAmoy.id]: http(`https://polygon-amoy.g.alchemy.com/v2/${ALCHEMY_KEY}``),
  }
})

createAppKit({
  ...generalConfig,
  adapters: [wagmiAdapter],
})

export function AppKitProvider(children) {
  return (
    <WagmiProvider config={wagmiAdapter.wagmiConfig}>
      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
    </WagmiProvider>
  )
}

```
Thank you for any help!

**Updated**
I found the way to override rpcUrls, but requests are still made from the WalletConnect url:

```
wagmiAdapter.wagmiConfig.chains.forEach(
  ({ id, rpcUrls }) => (rpcUrls.default.http = myRpcUrl)
)
```

![image](https://github.com/user-attachments/assets/26ab75a8-a1dc-4747-98f2-5be6571bae79)

![image](https://github.com/user-attachments/assets/ed5f5c4f-17ec-4abe-8035-3cc08d74d0c9)

Any ideas how I can use my rpcUrl?

**Comment by rtomas** [comment]
Are you usign latest version ?

**Reply by juli2a** [reply]
yes
```
"wagmi": "^2.12.32",
"@reown/appkit": "^1.5.0",
"@reown/appkit-adapter-wagmi": "^1.5.0",
```

**Reply by juli2a** [reply]
@rtomas I found that transports works on v1.3.2, it's broken since v1.4.0

**Reply by arein** [reply, marked as answer]
@juli2a we'll update you on this soon - meanwhile feel free to go ahead with the provided RPC and then update once you sorted it out 

also: https://docs.reown.com/appkit/react/core/custom-networks
